modules/distributor/distributor.go

```diff
@@ -300,6 +338,17 @@ func (d *Distributor) PushBatches(ctx context.Context, batches []*v1.ResourceSpa
 		recordDiscaredSpans(err, userID, spanCount)
 	}
 
+	if d.cfg.MetricsGeneratorEnabled && len(d.overrides.MetricsGeneratorProcessors(userID)) > 0 && err == nil {
+		// Handle requests sent to the metrics-generator in a separate goroutine, this way we don't
+		// influence the overall write
+		go func() {
```

Comment:
Already discussed but want to note the concerns about this goroutine. We can definitely merge as is, but I think long term we want a queue here. We will definitely want a way to limit the amount of pending requests to the metrics generators in case they are having issues and requests start backing up.

Reply:
Yep :+1: I'll add a section to the PR description to list these tasks. Once the PR is merged we can create dedicated issues for them.